### PR TITLE
[FE-42] Add before/after hooks around http retriever

### PR DESCRIPTION
**Pull request by zakharantoshchenko**
https://deliveroo.atlassian.net/browse/FE-42
Add before/after hooks for HttpRetriever#retrieve

### CODE REVIEW

File: lib/determinator/retrieve/http_retriever.rb [hunk 1]
```diff
@@ -10,11 +10,14 @@ def initialize(connection:)
       end
 
       def retrieve(name)
+        before_hook
         response = @connection.get("/features/#{name}")
+        after_hook(response.status)
```
Comment (me) [inline lib/determinator/retrieve/http_retriever.rb:15]: As discussed, can we pass in the error here? (nil in this case, and populating it inside the rescue).

**Review by me — COMMENTED** [review]
Looks great, just one comment.

Could you also update the readme with info about how to use these hooks? Thanks!

File: lib/determinator/retrieve/http_retriever.rb (hunk 1 above)
Comment (zakharantoshchenko) [reply lib/determinator/retrieve/http_retriever.rb:15]: done

File: README.md [hunk 2]
```diff
@@ -208,6 +208,26 @@ variant = determinator.which_variant(
 The `app_version` constraint for that flag needs to follow ruby gem version constraints. We support the following operators: `>, <, >=, <=, ~>`. For example:
 `app_version: ">=1.2.0"`
 
+### Using hooks for retriever
```
Comment (me) [inline README.md:211]: Should we move this a bit further up, under the `Using over http` section?

**Review by me — APPROVED** [review]
LGTM! One comment on the readme

File: README.md (hunk 2 above)
Comment (zakharantoshchenko) [reply README.md]: done

**Pull request merged**